js/modules/k6/ws/ws_test.go

```diff
@@ -504,3 +539,53 @@ func TestTLSConfig(t *testing.T) {
 	})
 	assertSessionMetricsEmitted(t, stats.GetBufferedSamples(samples), "", sr("WSSBIN_URL/ws-close"), 101, "")
 }
+
+func TestReadPump(t *testing.T) {
+	var closeCode int
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		conn, err := (&websocket.Upgrader{}).Upgrade(w, r, w.Header())
+		assert.NoError(t, err)
+		closeMsg := websocket.FormatCloseMessage(closeCode, "")
+		_ = conn.WriteControl(websocket.CloseMessage, closeMsg, time.Now().Add(time.Second))
+	}))
```

Comment:
You could've used the httpmultibin and added a custom handler as we do in https://github.com/loadimpact/k6/blob/4f04c7fa523ca46c9d48b3407d82d3edf33815cb/js/modules/k6/http/request_test.go#L157-L172 . This approach is definitely faster, and is okay this time, but if you didn't know you could I am sharing it for future cases ;)

Reply:
Thanks, I'm aware of `httpmultibin` (I rewrote the WS handlers in d34bce5a4 ;)), but I prefer not using it here since this is very specific to this particular test, as I needed to modify the close code returned by the server, and I didn't want to pollute or mess around with httpmultibin.

Reply:
I think @MStoykov meant that you can use `httpmultibin`, but define custom _local_ HTTP/WS handlers only in your particular test, no need to pollute the global one. But in any case, this is such a simple test that it wouldn't have made any difference, so I don't think it matters.